File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.js
function solution(operations) {
    const pq = [];
    
    for(const operation of operations){
        const [op, num] = operation.split(" ");
        
        if(op==='I'){
            pq.push(Number(num));
            pq.sort((a,b)=>a-b);
        }
        
        else{
            if(num==='1')   pq.pop();
            else    pq.shift();
        }
    }
    
    return pq.length===0 ? [0,0] : [pq[pq.length-1], pq[0]];
}

File: 프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.py
from collections import deque

def solution(operations):
    queue = deque()
    
    for operation in operations:
        op, num = operation.split(' ')
        num = int(num)
        
        if op == 'I':
            queue.append(num)
        elif op == 'D':
            if not queue:
                continue
            queue = deque(sorted(queue))
            if num == 1:
                queue.pop()
            else:
                queue.popleft()
    
    queue = deque(sorted(queue))
    
    return [0, 0] if not queue else [queue[-1], queue[0]]